### PR TITLE
fix outbound event payloads and handle 4xx errors

### PR DESCRIPTION
**Pull request by ajwootto**

### CODE REVIEW

File: eventqueue.go [hunk 1]
```diff
@@ -181,6 +183,25 @@ func (e *EventQueue) FlushEvents() (err error) {
...
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+
+			responseBody, err := ioutil.ReadAll(resp.Body)
+			resp.Body.Close()
+
+			if err != nil {
+				log.Println(err)
```
Comment (JamieSinn) [inline eventqueue.go:197]:
```suggestion
				printf(err)
```

File: eventqueue.go [hunk 2]
```diff
@@ -181,6 +183,25 @@ func (e *EventQueue) FlushEvents() (err error) {
 			continue
 		}
 
+		if resp.StatusCode >= 400 {
+			err = e.localBucketing.onPayloadFailure(payload.PayloadId, false)
+			if err != nil {
+				log.Println(err)
```
Comment (JamieSinn) [inline eventqueue.go:189]:
```suggestion
				printf(err)
```

File: eventqueue.go [hunk 3]
```diff
@@ -181,6 +183,25 @@ func (e *EventQueue) FlushEvents() (err error) {
...
+
+			responseBody, err := ioutil.ReadAll(resp.Body)
+			resp.Body.Close()
+
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+
+			log.Println("Error sending events", string(responseBody))
```
Comment (JamieSinn) [inline eventqueue.go:201]:
```suggestion
			printf("Error sending events", string(responseBody))
```

**Review by JamieSinn — CHANGES_REQUESTED** [review]

File: eventqueue.go (hunk 1 above)
Comment (ajwootto) [reply eventqueue.go:197]: that doesnt match any of the other logging in this file

**Review by JamieSinn — APPROVED** [review]

**Pull request merged**